noodles-bed/src/de.rs

```diff
@@ -1,934 +1,296 @@
-                    chrom = Some(map.next_value()?);
-                }
-                Field::Start => {
-                    if start.is_some() {
-                        return Err(de::Error::duplicate_field("start"));
-                    }
-                    start = Some(map.next_value()?);
-                }
-                Field::End => {
-                    if end.is_some() {
-                        return Err(de::Error::duplicate_field("end"));
-                    }
-                    end = Some(map.next_value()?);
-                }
-            }
-        }
-        let chrom: &str = chrom.ok_or_else(|| de::Error::missing_field("chrom"))?;
-        let start: usize = start.ok_or_else(|| de::Error::missing_field("start"))?;
-        let end: usize = end.ok_or_else(|| de::Error::missing_field("end"))?;
-
-        Record::<3>::builder()
-            .set_reference_sequence_name(chrom)
-            .set_start_position(Position::try_from(start).map_err(|err| de::Error::custom(err))?)
-            .set_end_position(Position::try_from(end).map_err(|err| de::Error::custom(err))?)
-            .build().map_err(|err| de::Error::custom(err))
-    }
-}
-
-
-impl<'de> Deserialize<'de> for Record<3> {
-    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        deserializer.deserialize_struct("Record", FIELDS, Record3Visitor)
-    }
-}
-
-pub fn from_reader<R>(reader: &mut Reader<R>) -> std::result::Result<Vec<Record<3>>, std::io::Error>
-where
-    R: BufRead
-{
-    reader
-        .records::<3>()
-        .collect()
-}
-
-pub struct Deserializer<'de> {
-    // This string starts with the input data and characters are truncated off
-    // the beginning as data is parsed.
+pub struct RecordDeserializer<'de> {
     input: &'de str,
 }
 
-impl<'de> Deserializer<'de> {
-    // By convention, `Deserializer` constructors are named like `from_xyz`.
-    // That way basic use cases are satisfied by something like
-    // `serde_json::from_str(...)` while advanced use cases that require a
-    // deserializer can make one with `serde_json::Deserializer::from_str(...)`.
-    pub fn from_str(input: &'de str) -> Self {
-        Deserializer { input }
-    }
-}
-
-// By convention, the public API of a Serde deserializer is one or more
-// `from_xyz` methods such as `from_str`, `from_bytes`, or `from_reader`
-// depending on what Rust types the deserializer is able to consume as input.
-//
-// This basic deserializer supports only `from_str`.
-pub fn from_str<'a, T>(s: &'a str) -> Result<T>
+fn from_str<'a, T>(s: &'a str) -> Result<T>
 where
     T: Deserialize<'a>,
 {
-    let mut deserializer = Deserializer::from_str(s);
+    let mut deserializer = RecordDeserializer::from_str(s);
     let t = T::deserialize(&mut deserializer)?;
     if deserializer.input.is_empty() {
         Ok(t)
     } else {
-        Err(Error::TrailingCharacters)
+        panic!()
```

Comment:
Return an `Err` here. For example `Err(ser::Error::custom("trailing characters"))`.